pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -33,6 +34,153 @@ const (
 	AnnImportDiskId = "cdi.kubevirt.io/storage.import.volumeId"
 )
 
+// OS types
+const (
+	Linux   = "linux"
+	Windows = "windows"
+)
+
+// Secure boot options
+const (
+	Required = "required"
+	Disabled = "disabled"
+	Optional = "optional"
+)
+
+// Machine types
+const (
+	PC  = "pc"
+	Q35 = "q35"
+)
+
+// Firmware types
+const (
+	BIOS = "bios"
+	EFI  = "uefi"
+)
+
+// CPU Policies
+const (
+	CPUPolicyDedicated = "dedicated"
+	CPUPolicyShared    = "shared"
+)
+
+// CPU Thread policies
+const (
+	CPUThreadPolicyPrefer  = "prefer"
+	CPUThreadPolicyIsolate = "isolate"
+	CPUThreadPolicyRequire = "require"
+)
+
+// Bus types
+const (
+	VirtioSCSI = "virtio_scsi"
+	VirtIO     = "virtio"
+	SATA       = "sata"
+	SCSI       = "scsi"
+	IDE        = "ide"
+	USB        = "usb"
+)
+
+// Input types
+const (
+	Tablet    = "tablet"
+	USBTablet = "usbtablet"
+)
+
+// Video models
+const (
+	VGA    = "vga"
+	Cirrus = "cirrus"
+	VMVGA  = "vmvga"
+	Xen    = "xen"
+	QLX    = "qxl"
+	Gop    = "gop"
+	None   = "none"
+	Bochs  = "bochs"
+)
+
+// Disk Formats
+const (
+	AMI   = "ami"
+	ARI   = "ari"
+	AKI   = "aki"
+	VHD   = "vhd"
+	VHDX  = "vhdx"
+	VMDK  = "vmdk"
+	RAW   = "raw"
+	QCOW2 = "qcow2"
+	VDI   = "vdi"
+	PLOOP = "ploop"
+	ISO   = "iso"
+)
+
+// Image Properties
+const (
+	Arch            = "architecture"
+	HypervisorType  = "hypervisor_type"
+	CPUPolicy       = "hw_cpu_policy"
+	CPUThreadPolicy = "hw_cpu_thread_policy"
+	CPUCores        = "hw_cpu_cores"
+	CPUSockets      = "hw_cpu_sockets"
+	CPUThreads      = "hw_cpu_threads"
+	FirmwareType    = "hw_firmware_type"
+	MachineType     = "hw_machine_type"
+	CDROMBus        = "hw_cdrom_bus"
+	PointerModel    = "hw_pointer_model"
+	VideoModel      = "hw_video_model"
+	DiskBus         = "hw_disk_bus"
+	VIFModel        = "hw_vif_model"
+	OSType          = "os_type"
+	OSDistro        = "os_distro"
+	OSVersion       = "os_version"
+	OSSecureBoot    = "os_secure_boot"
+	HWVideoRAM      = "hw_video_ram"
+)
+
+// Flavor ExtraSpecs
+const (
+	FlavorSecureBoot    = "os:secure_boot"
+	FlavorCPUCores      = "hw:cpu_cores"
+	FlavorCPUSockets    = "hw:cpu_sockets"
+	FlavorCPUThreads    = "hw:cpu_threads"
+	FlavorMaxCPUCores   = "hw:max_cpu_cores"
+	FlavorMaxCPUSockets = "hw:max_cpu_sockets"
+	FlavorMaxCPUThreads = "hw:max_cpu_threads"
+	FlavorHWVideoRAM    = "hw_video:ram_max_mb"
+)
+
+// Network types
+const (
+	Pod    = "pod"
+	Multus = "multus"
+)
+
+// Default CPU parameters
+const (
+	DefaultCPUSockets = "1"
+	DefaultCPUCores   = "1"
+	DefaultCPUThreads = "1"
+)
+
+// Default properties
```

Comment:
default for the resources we get from openstack, right?

Reply:
Yes, those are the observed defaults for VMs when the image they were created from has no properties configured.